### PR TITLE
feat: gpkg creation integration

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**

### CODE REVIEW

File: MergerCli/SourceParser.cs [hunk 1]
```diff
@@ -79,14 +90,47 @@ private IData ParseFileSource(string[] args, ref int idx, int batchSize, bool is
...
+        {
+            const int requiredParamCount = 2;
+            const int optionalParamCount = 4;
+            int paramCount = this.ValidateAndGetSourceLength(args, idx, requiredParamCount, optionalParamCount);
+            string sourceType = args[idx];
+            string sourcePath = args[idx + 1];
+            bool isOneXOne = false;
+            GridOrigin? origin = null;
+            Extent? extent = null;
+            int? maxZoom = null;
```
Comment (shimoncohen) [inline MergerCli/SourceParser.cs:103]: Add comment - this affects only the `gpkg_tile_matrix` table.

File: MergerLogic/DataTypes/Gpkg.cs [hunk 2]
```diff
@@ -43,37 +41,43 @@ public class Gpkg : Data<IGpkgUtils>
             else
             {
                 this._coordsFromCurrentGrid = cords => cords;
-            };
+            }
+            if (!this.utils.Exist() && isBase && maxZoom is not null && extent is not null)
+            {
+                // extent and tile matrix will be properly set after merge
```
Comment (shimoncohen) [inline MergerLogic/DataTypes/Gpkg.cs:47]: Remove comment.

File: MergerLogic/DataTypes/Gpkg.cs [hunk 3]
```diff
@@ -127,7 +131,7 @@ protected override Tile GetLastExistingTile(Coord baseCoords)
             }
 
             Tile lastTile = this.utils.GetLastTile(coords, baseCoords);
-            return this._toCurrentGrid(lastTile);
+            return lastTile is not null ? this._toCurrentGrid(lastTile) : null;
```
Comment (shimoncohen) [inline MergerLogic/DataTypes/Gpkg.cs:134]: Change to `if`.

File: MergerLogic/Utils/GpkgUtils.cs [hunk 4]
```diff
@@ -553,16 +579,29 @@ private void CreateSqureGrid(SQLiteConnection connection, int maxZoom, int baseW
...
         {
             //TODO: add support for 1x1? (copy this function and change grid bbox and base zoom parameters)
             using (var command = connection.CreateCommand())
             {
                 command.CommandText = "INSERT INTO \"gpkg_tile_matrix_set\" VALUES " +
-                    $"({this._tileCache},4326,-180,-90,180,90);";
+                    $"('{this._tileCache}',4326,-180,-90,180,90);";
+                command.ExecuteNonQuery();
+            }
+            CreateSqureGrid(connection, maxZoom, 2, 1, 0.703125, 2, 256);//creates 2X1 grid
```
Comment (shimoncohen) [inline MergerLogic/Utils/GpkgUtils.cs:592]: Make all magic numbers as constants.

Comment (shimoncohen) [reply MergerLogic/Utils/GpkgUtils.cs:592]: Some should be in `geoUtils`.

File: MergerLogic/Utils/GpkgUtils.cs [hunk 5]
```diff
@@ -553,16 +579,29 @@ private void CreateSqureGrid(SQLiteConnection connection, int maxZoom, int baseW
...
 
+
         private void Add2X1Data(SQLiteConnection connection, int maxZoom)
         {
             //TODO: add support for 1x1? (copy this function and change grid bbox and base zoom parameters)
             using (var command = connection.CreateCommand())
             {
                 command.CommandText = "INSERT INTO \"gpkg_tile_matrix_set\" VALUES " +
-                    $"({this._tileCache},4326,-180,-90,180,90);";
+                    $"('{this._tileCache}',4326,-180,-90,180,90);";
```
Comment (shimoncohen) [inline MergerLogic/Utils/GpkgUtils.cs:589]: Same, make `4326` constant in `geoUtils`.

**Review by shimoncohen — CHANGES_REQUESTED** [review]

File: MergerLogic/DataTypes/Gpkg.cs (hunk 3 above)
Comment (cl-shaharshavit-rnd) [reply MergerLogic/DataTypes/Gpkg.cs:134]: done :(

File: MergerLogic/DataTypes/Gpkg.cs (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply MergerLogic/DataTypes/Gpkg.cs:47]: done

File: MergerCli/SourceParser.cs (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply MergerCli/SourceParser.cs:103]: done

**Review by shimoncohen — APPROVED** [review]

**Pull request merged**